pkg/edit/store_api.go

```diff
@@ -10,37 +10,88 @@ import (
 	"src.elv.sh/pkg/eval"
 	"src.elv.sh/pkg/eval/vals"
 	"src.elv.sh/pkg/parse/parseutil"
+	"src.elv.sh/pkg/store"
 )
 
 var errStoreOffline = errors.New("store offline")
 
 //elvdoc:fn command-history
 //
-// Outputs the entire command history as a stream of maps. Each map has a `id`
-// key that identifies the sequence number of the entry, and a `cmd` key that
-// identifies the content.
+// ```elvish
+// edit:command-history &dedup &newest-first
+// ```
 //
-// Use indexing to extract individual entries. For example, to extract the
-// content of the last command, do this:
+// Outputs the command history as a stream of maps unless `&cmd-only` is true. Each map has a `id`
+// key for the sequence number of the command, and a `cmd` key for the text of the command. If
+// `&dedup` is true only the most recent instance of each command (when comparing just the `cmd`
+// key) is output. If `&newest-first` is true the output is in newest to oldest order (default is
+// oldest to newest). If `&cmd-only` is true then only the command text is output (default is a map
+// as described previously).
+//
+// Use indexing to extract individual entries. For example, to extract the text of the most recent
```

Comment:
This is a bit of an out-of-place comment here since it applies to all commands that output a stream of values. It's sufficient to just say that the following are useful examples.